### ISSUE TITLE
[a11y]: Tabs [dismissible] - unable to dismiss in iPhone

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#14820]
### Package

@carbon/react

### Browser

Safari & iPhone

### Package version

@carbon/react@1.38.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS & VO

### Description
"No way of interaction available for VO users to delete A tab in iPhone.

NOTE: In call its understood there is delete icon which helps the visual users to delete the TAB but its not read by Voiceover."

### Recommended Fix
Provide Delete Name and Button Role for Delete ICON and make it keyboard operable

When the standard interaction is not possible, user is unable to accomplish A task.

### WCAG 2.1 Violation

4.1.2 Name, Role, Value

**Comment by alisonjoseph** [comment]
Question for @kingtraceyj as I believe you worked on dismissable tabs initially. Or any @carbon-design-system/design that wants to chime in. 

Currently the only way to access the delete icon by keyboard is by using the delete key, this obviously doesn't work on phone/tablet devices. 

Would you be opposed to moving the delete action into the tab order for keyboard? So a keyboard user would tab to focus the tab, then tab again to focus the delete icon, then if they tab again it would go into the content, or if they pressed the right/left arrows it would switch the tab focus the same way it does now?

**Comment by mbgower** [comment]
There seems to be an easy fix to the discoverability problem, BTW. The div containing the tab item has aria-hidden="true" on it. Remove that, and the announcement "Press delete to close tab" is announced by the screen reader.
The svg has aria-hidden set to false, but because it is a child of the div, it is inheriting the hidden attribute.
<img width="371" alt="image" src="https://github.com/carbon-design-system/carbon/assets/14298245/61aee11a-9762-4d46-8ff5-7fdb40392aa5">

**Comment by alisonjoseph** [comment]
@mbgower oh yea, that part seems like something we could easily fix with aria-hidden. That still wouldn't allow a user using voiceover on iphone to dismiss it though, since they don't have a delete key to press?

**Comment by mbgower** [comment]
Yep, agreed, that this idea of the delete key is interesting -- for those iOS VoiceOver users who do not use a bluetooth keyboard. But that's also a consideration beyond just the dismissable tabs. For instance, there are lots of popups that don't have dismiss functions in the tab order, which users close with Esc. A touch user dismisses by clicking outside the overlay. How does a iOS VoiceOver user dismiss those?

**Comment by mbgower** [comment]
Here's what the ARIA Authoring Practices says in their [tabs pattern](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/):
> Delete (Optional): If deletion is allowed, deletes (closes) the current tab element and its associated tab panel, sets focus on the tab following the tab that was closed, and optionally activates the newly focused tab. If there is not a tab that followed the tab that was deleted, e.g., the deleted tab was the right-most tab in a left-to-right horizontal tab list, sets focus on and optionally activates the tab that preceded the deleted tab. If the application allows all tabs to be deleted, and the user deletes the last remaining tab in the tab list, the application moves focus to another element that provides a logical work flow. As an alternative to Delete, or in addition to supporting Delete, the delete function is available in a context menu.

**Comment by mbgower** [comment]
https://github.com/w3c/aria-practices/issues/2973

**Comment by mbgower** [comment]
I got a response back form the aria-practices working group
> this type of pattern is addressed by the new aria-actions property that should be in ARIA 1.4 and is a 2024 priority for the ARIA WG.
The APG TF is looking for an engineer who wants to be part of the team to build our first example that uses aria-actions. Both this tabs example and tree are strong candidates. Thank you for raising this idea.
